src/hooks/useGridKeyboardNavigation/useGridKeyboardNavigation.js

```diff
@@ -0,0 +1,106 @@
+import { useCallback, useLayoutEffect, useState } from "react";
+import useFullKeyboardListeners from "../useFullKeyboardListeners";
+// import { GridKeyboardNavigationContext } from "../../components/GridKeyboardNavigation/GridKeyboardNavigationContext";
+import { calcActiveIndexAfterArrowNavigation, getActiveIndexFromInboundNavigation } from "./gridKeyboardNavigationHelper";
+import useEventListener from "../useEventListener";
+
+const NO_ACTIVE_INDEX = -1;
+
+/**
+ * @typedef useGridKeyboardNavigationResult
+ * @property {number} activeIndex - the currently active index
+ * @property {function} onSelectionAction - the callback which should be used to select an item. It should be called with the selected item's index. Use this callback for onClick handlers, for example.
+ */
+
+/**
+ * A hook which is used for accessible keyboard navigation. Useful for components rendering a list of items that can be navigated and selected with a keyboard.
```

Comment:
mind blown! @laviomri 
love JSDocs!

Reply:
glad you like it! 
if only there was a way to sync between the JSDoc and the documentation in the stories (like the `Returns`/`Arguments` sections), it would have been awesome